### DISCUSSION TITLE
Roku Beta Testing Group

### DISCUSSION THREAD

**Discussion by PRProd** [PRProd/HA-Firemote#275] [Announcements]
# Upcoming Roku Support
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/4c747f1e-023f-4b23-9f69-8c7d0e14885c)

I'd love to have YOU as a beta tester! Follow this thread and you'll be notified right away when new versions are ready, and you'll be first in line to share your input before launch! (Tagging @ThatGuy9811 , @darrenhomme as known interested parties)

Please add your input here, or in the discord group: https://discord.gg/sqfR2hCAFz

## To Do List
 - [x] Release first beta version for testing
 - [x] Receive and evaluate community feedback
 - [x] Create generic remote for TCL / Roku TV
 - [x] Create generic remote for Westinghouse / Roku TV
 - [x] Create generic remote for Hisense / Roku TV
 - [x] Update Documentation
 - [x] Release

## Known Limitations / Issues
 - Volume, channel, mute, find remote, and other buttons are available as functions in the **App Launch Button** dropdown menus as opposed to putting some of those controls on the side of the remote where their tap targets would just be too small.
 - Current Home Assistant Roku Integration does not seem to support the "literal" character input as advertised.  This prevents a keyboard integration for now.  [Issue 105776](https://github.com/home-assistant/core/issues/105776) has been opened in Home Assistant Core
 - Current Home Assistant Roku Integration seems to have broken support for the Roku Search feature
   -  [Issue 102024](https://github.com/home-assistant/core/issues/102024) was opened and closed for this without any changes made
   - [Issue 105827](https://github.com/home-assistant/core/issues/105827) was just opened to hopefully move it along
 - Current Home Assistant Roku Integration seems to not have a "sleep" function which is included on some TV remotes
 - Current Home Assistant Roku Integration and Roku API does not appear to have support for the programmable **1** and **2** buttons on the voice remote pro.

**Comment by PRProd** [comment]
### Beta 1 has been released
Good news, testers!  Firemote Beta Version [v3.2.0b1](https://github.com/PRProd/HA-Firemote/releases/tag/v3.2.0b1) is now available to download and test!  I'd love to hear from you!

**Comment by darrenhomme** [comment]
I have v3.2.0b1 installed and will start testing tonight

**Comment by darrenhomme** [comment]
![image](https://github.com/PRProd/HA-Firemote/assets/18243349/bc489bb9-9fcb-448a-8bb1-3c4fce9b90d9)
I'm still seeing a console error when selecting device family

**Reply by PRProd** [reply]
Can you try to re-copy the lit directory from the repo into your HA instance?

**Comment by darrenhomme** [comment]
![image](https://github.com/PRProd/HA-Firemote/assets/18243349/95bffd3d-06a0-406b-9b0e-c8293de5137e)
![image](https://github.com/PRProd/HA-Firemote/assets/18243349/336b7d62-f7a6-4a8f-8536-99df9a0616bd)
Do I need to change permissions, user, or groups?

**Reply by PRProd** [reply]
I don't think so??

**Reply by PRProd** [reply]
I might know... there was a bug I fixed in the main release earlier today.  Can you try that version to see if the console error persists?

**Reply by darrenhomme** [reply]
I'll grab that. Was it related to this? 
![image](https://github.com/PRProd/HA-Firemote/assets/18243349/ec969aea-8446-47a1-ac61-98563d671eba)

**Reply by PRProd** [reply]
No

**Reply by darrenhomme** [reply]
![image](https://github.com/PRProd/HA-Firemote/assets/18243349/738f00fc-25e6-44ce-a83e-3bb2f4dc330f)

**Reply by darrenhomme** [reply]
I deleted HA-Firemote.js and lit before copying 3.1.6 in

**Reply by darrenhomme** [reply]
Going back to the beta version

**Comment by darrenhomme** [comment]
I tested the channel up and down buttons with the antenna input on the tcl roku tv. the tv tuner app launch button doesn't start the live tv app for me. is there a live tv option?

**Reply by PRProd** [reply]
The [Roku Integration Docs](https://www.home-assistant.io/integrations/roku#remote) say that there should be a "input_tuner" remote command that I assume is meant to do this task.  Pressing the "TV Tuner" launch button issues this remote command right now.  If it doesn't make your TV switch to the Antenna/Cable tuner mode, I'm not sure what that command is supposed to do.  We might want to do another remote session

**Comment by PRProd** [comment]
## Beta 2 Has been released
Firemote Beta version [v3.2.0b2](https://github.com/PRProd/HA-Firemote/releases/tag/v3.2.0b2) has been released for testing.  

@darrenhomme ~ Please let me know if this solves the console log error message issue you reported.  I wasn't able to reproduce it on my end, so I'm unable to confirm a fix without your input.

**Reply by darrenhomme** [reply]
Console error is not present in b2

**Comment by manonstreet** [comment]
So far this beta is working great!  I agree, it would be awesome if HA fixed the remote being unable to send literals.  That wold unlock so many cool features.

@PRProd I'm not sure this is related to the roku build, or a more generalized issue (let me know and I can request elsewhere).  Using the `script` keyword within a custom launcher does not respect passing variables via the data parameter.  This can be worked around by instead calling `service: script.turn_on` with the required fields.  But it would be cleaner config if it worked via script.

**Reply by PRProd** [reply]
Thanks for the tip @manonstreet!  If you'd like to open a new [feature request](https://github.com/PRProd/HA-Firemote/issues/new?assignees=PRProd&labels=enhancement&projects=&template=feature_request.yaml&title=%5BFeature+Request%5D%3A+) that might help me stay organized better.  I'll try to get to it soon

**Reply by PRProd** [reply]
@manonstreet - I saw a pull request against the Home Assistant core come through that should fix the literals issue.  :rocket: :tada:

Have a look at the [Known Limitations](https://github.com/PRProd/HA-Firemote/wiki#known-limitations) section in the [Firemote wiki](https://github.com/PRProd/HA-Firemote/wiki) under the [Roku](https://github.com/PRProd/HA-Firemote/wiki#roku) heading for more info.

**Reply by PRProd** [reply]
The keyboard functionality works now as of Firemote version [v3.2.3](https://github.com/PRProd/HA-Firemote/releases/tag/v3.2.3) running on Home Assistant 2024.2.0 or newer!  :tada:

**Reply by manonstreet** [reply]
Keyboard functionality works! ... mostly.  It seems you can't send the @ literal.  I like to use the keyboard option for logging in to Roku apps by copying and pasting the user/pass out of my password manager.  No @ unfortunately means email address can't be pasted :/

**Reply by PRProd** [reply]
Well... heck.  I'll have to raise another issue I think.  The ampersand might not work either...

**Reply by PRProd** [reply]
In total, I found 7 characters that did not work for me.  I opened [issue #110511](https://github.com/home-assistant/core/issues/110511) for help.

**Reply by PRProd** [reply]
@manonstreet & @darrenhomme ~

A fix for this should be included in Home Assistant 2024.2.3.  Pull Request details can be found at [PR #110670](https://github.com/home-assistant/core/pull/110670)

**Comment by manonstreet** [comment]
The limitations above note that:
> Volume, channel, mute, find remote, and other buttons are available as functions in the dropdown menu as opposed to putting some of those controls on the side of the remote.

Can you please shed some light on this?  I don't see any dropdown menu options on the card or in the documentation.

**Reply by manonstreet** [reply]
Nevermind!  I understand what is meant.  When configuring the card, the drop down for one of the custom app launch buttons can be volume, mute, etc.

**Reply by PRProd** [reply]
I updated the language to be a bit clearer.  Thanks, @manonstreet!

**Comment by PRProd** [comment]
## Release Candidate is available
Firemote beta version [3.2.0b6](https://github.com/PRProd/HA-Firemote/releases/tag/v3.2.0b6) is now available for download.

I expect to push this into the official version of Firemote very soon.  I'm going to chip away at some documentation first though.  I want to get ahead of the "hey!  you should add a keyboard!" and the "why doesn't the sleep button work?" reactions.  :smile:

If any of you find a lingering issue that hasn't been addressed yet, please let me know.  Otherwise, I think it's good to go!

**Comment by PRProd** [comment]
# Roku Support has been released!
Official Firemote version [v3.2.0](https://github.com/PRProd/HA-Firemote/releases/tag/v3.2.0) has just been published!  Thank you everyone for all of your help!

This discussion will remain open for a while.  If you find anything wrong, feel free to mention it here or to open an issue.

**Comment by jonezy35** [comment]
It seems the volume button functions aren't working for me. I was able to put them on the remote, but they do nothing. Anyone else experiencing this?

**Reply by PRProd** [reply]
Hey @jonezy35!

Did you get a chance to look at the entry in the readme [FAQ section](https://github.com/PRProd/HA-Firemote?tab=readme-ov-file) titled [Why won't the volume, mute, and/or power buttons work from my Firemote card?](https://github.com/PRProd/HA-Firemote?tab=readme-ov-file#why-wont-the-volume-mute-andor-power-buttons-work-from-my-firemote-card)

**Reply by jonezy35** [reply]
Hey @PRProd! Thank you for the reply!

I did and maybe I'm misunderstanding, but that is referring to my samsung TV settings correct? My Roku remote can already control the volume, and from my research enabling CEC on a roku streaming stick is done by going to the system settings and checking 1-touch-play which is already checked for me.

**Reply by jonezy35** [reply]
Also I mean to add CEC is already enabled on my Samsung TV as well

**Reply by PRProd** [reply]
Some TVs only have CEC/eARC on select HDMI inputs.  Is it possible that your stick is plugged into an input that doesn't support it?

**Reply by jonezy35** [reply]
@PRProd it could be possible, but doesn't the fact that my regular Roku remote can control volume mean that CEC is working? Otherwise my physical Roku remote wouldn't be able to control volume?

**Reply by PRProd** [reply]
It is likely that your Roku remote controls your TV's volume through an infrared signal and not through CEC.  Does your physical remote have an IR emitter on the front?

![image](https://github.com/PRProd/HA-Firemote/assets/25486599/6c9c8e33-3a8c-48ec-8140-2b7879ea252e)
Somewhere inside that shiny section is where it lives.  The physical appearance may vary from remote to remote, this is simply a photo I took of what mine looks like.  Additional information about IR vs CEC can be found in [this Firemote Wiki article](https://github.com/PRProd/HA-Firemote/wiki/Using-CEC-Controls).

That being said, there is a different angle you can take to address this issue.  Because of [the way Firemote works](https://github.com/PRProd/HA-Firemote/wiki#how-does-firemote-send-commands-to-my-devices), this suggestion will only work if your Samsung TV's volume is able to be controlled directly from Home Assistant.  If you **can** control your TV's volume through Home Assistant, then the contents of [this discussion - #314](https://github.com/PRProd/HA-Firemote/discussions/314) will be able to guide you in the right direction.

**Reply by jonezy35** [reply]
Thank you! I can control the volume through home assistant, so I will give that a look after work today!

**Reply by jonezy35** [reply]
Update, [this](https://github.com/PRProd/HA-Firemote/discussions/314#discussioncomment-8323102) worked! Thank you so much!!
